tests/frameworks/OSXShims/src/LoggingNative.mm

```diff
@@ -15,44 +15,92 @@
 //******************************************************************************
 
 #import <LoggingNative.h>
+#import <Starboard/SmartTypes.h>
 #import <Foundation/NSString.h>
 
 #import <cstdarg>
 
+// NOTE: this function is assumed to only run in Little Endian environments like OSX
+static NSStringEncoding _getWcharEncoding() {
+    switch (sizeof(wchar_t)) {
+        case 2:
+            return NSUTF16LittleEndianStringEncoding;
+        case 4:
+            return NSUTF32LittleEndianStringEncoding;
+        default:
+            assert(false);
+    }
+}
+
+static NSStringEncoding getWcharEncoding() {
+    static NSStringEncoding encoding = _getWcharEncoding();
+    return encoding;
+}
+
 void TraceVerbose(const wchar_t* tag, const wchar_t* format, ...) {
-    NSString* formatString = [NSString stringWithFormat:@"[%ls] %ls", tag, format];
+    StrongId<NSString> nsTag;
```

Comment:
@bbowman ARC is per-file, though, not per-module. Additionally, `.attach` is invalid in ARC mode, since `StrongId` resolves to `__strong id`.

Our options here are minimal; either we use `stringWithCString:encoding:` which does not allow us to specify length, or we use `autorelease` and make 100% certain that this file is built with ARC off. My vote is for the second, since it makes it significantly easier to reason about what's happening.